patches/net/minecraft/data/models/model/ModelTemplate.java.patch

```diff
@@ -9,43 +9,99 @@
++    }
++
 +    // Neo: Use ExistingFileHelper sensitive version
 +    @Deprecated
      public ResourceLocation createWithOverride(
          Block p_125617_, String p_125618_, TextureMapping p_125619_, BiConsumer<ResourceLocation, Supplier<JsonElement>> p_125620_
      ) {
-         return this.create(ModelLocationUtils.getModelLocation(p_125617_, p_125618_), p_125619_, p_125620_);
-     }
- 
+-        return this.create(ModelLocationUtils.getModelLocation(p_125617_, p_125618_), p_125619_, p_125620_);
+-    }
+-
++        return this.createWithOverride(p_125617_, p_125618_, p_125619_, p_125620_, null);
++    }
++
++    public ResourceLocation createWithOverride(
++            Block p_125617_, String p_125618_, TextureMapping p_125619_, BiConsumer<ResourceLocation, Supplier<JsonElement>> p_125620_, @org.jetbrains.annotations.Nullable net.neoforged.neoforge.common.data.ExistingFileHelper fileHelper
++    ) {
++        return this.create(ModelLocationUtils.getModelLocation(p_125617_, p_125618_), p_125619_, p_125620_, fileHelper);
++    }
++
 +    // Neo: Use ExistingFileHelper sensitive version
 +    @Deprecated
      public ResourceLocation create(ResourceLocation p_125613_, TextureMapping p_125614_, BiConsumer<ResourceLocation, Supplier<JsonElement>> p_125615_) {
-         return this.create(p_125613_, p_125614_, p_125615_, this::createBaseTemplate);
+-        return this.create(p_125613_, p_125614_, p_125615_, this::createBaseTemplate);
++        return this.create(p_125613_, p_125614_, p_125615_, (net.neoforged.neoforge.common.data.ExistingFileHelper) null);
++    }
++
++    public ResourceLocation create(ResourceLocation p_125613_, TextureMapping p_125614_, BiConsumer<ResourceLocation, Supplier<JsonElement>> p_125615_, @org.jetbrains.annotations.Nullable net.neoforged.neoforge.common.data.ExistingFileHelper fileHelper) {
++        return this.create(p_125613_, p_125614_, p_125615_, (p_266987_, p_266933_) -> createBaseTemplate(p_266987_, p_266933_, fileHelper));
      }
-@@ -60,6 +_,8 @@
+ 
+     public ResourceLocation create(
+@@ -60,12 +_,30 @@
          return p_266990_;
      }
  
 +    // Neo: Use ExistingFileHelper sensitive version
 +    @Deprecated
      public JsonObject createBaseTemplate(ResourceLocation p_266830_, Map<TextureSlot, ResourceLocation> p_266912_) {
++        return this.createBaseTemplate(p_266830_, p_266912_, null);
++    }
++
++    public JsonObject createBaseTemplate(ResourceLocation p_266830_, Map<TextureSlot, ResourceLocation> p_266912_, @org.jetbrains.annotations.Nullable net.neoforged.neoforge.common.data.ExistingFileHelper fileHelper) {
          JsonObject jsonobject = new JsonObject();
-         this.model.ifPresent(p_176461_ -> jsonobject.addProperty("parent", p_176461_.toString()));
+-        this.model.ifPresent(p_176461_ -> jsonobject.addProperty("parent", p_176461_.toString()));
```

Comment:
Instead of modifying the two lambdas, I would do the two checks at the top of the method:
```java
if (fileHelper != null) {
    this.model.ifPresent(parent -> Preconditions.checkState(fileHelper.exists(parent, ModelProvider.MODEL), "Model at %s does not exist", parent));
    p_266912_.forEach((slot, name) -> Preconditions.checkState(fileHelper.exists(name, ModelProvider.TEXTURE), "Texture %s does not exist in any known resource pack", name));
}
```